src/bootstrap/build/sanity.rs

```diff
@@ -49,12 +54,29 @@ pub fn check(build: &mut Build) {
         }
         need_cmd("cmake".as_ref());
         if build.config.ninja {
-            need_cmd("ninja".as_ref())
+            need_cmd("ninja".as_ref());
         }
         break
     }
 
-    need_cmd("python".as_ref());
+    let python = need_cmd("python".as_ref());
+
+    // On OSX we can only use the python from /usr/bin because otherwise it's
+    // incompatible with the LLDB python framework we load during the LLDB
+    // tests. It's somewhat common for a homebrew-installed python to exist, but
+    // we're unfortunately incompatible with that.
+    if build.config.build.ends_with("apple-darwin") {
+        if !python.starts_with("/usr/bin") {
+            panic!("
+
+Python detected ({}) is not /usr/bin/python and is unlikely to be able
+to run lldb tests. If this Python is installed via Homebrew then you'll likely
+need to uninstall it entirely (or move it out of 'PATH') to get lldb tests
+working.
+
+", python.display());
```

Comment:
Ok, so to be clear scary errors are printed out if `/usr/local/bin` is in `PATH` even if `/usr/bin/python` is used:

```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Applications/Xcode.app/Contents/SharedFrameworks/LLDB.framework/Resources/Python/lldb/__init__.py", line 98, in <module>
    import six
ImportError: No module named six
```

That being said the `lldb_batchmode.py` script seems to run just fine and continue despite this error.

Unfortunately I still can't get tests passing due to https://github.com/rust-lang/rust/issues/33062, but it seems like yes, we should just tell compiletest to use /usr/bin/python unconditionally.